addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1175,8 +1175,7 @@ def channel_join(self):
         self.add_members(self.env.user.partner_id.ids)
 
     @api.model
-    @api.returns("self", lambda channels: Store(channels).get_result())
-    def channel_create(self, name, group_id):
+    def _channel_create(self, name, group_id):
```

Comment:
This should be `_create_channel`, the `channel` here in the name meaning the channel type (similar to `group`) and not the channel model.
```suggestion
    def _create_channel(self, name, group_id):
```